### PR TITLE
[protobuf-c] Fix tool protoc-gen-c crash

### LINKED ISSUE #10745: protobuf-c code generation using protoc-gen-c is not working
I installed protobuf-c[tools] package in x64-linux env and build protoc-gen-c successfully.
But, when I try generate c code using protoc and plugin built failed.
FYI) If I built protobuf-c not using vcpkg, code generation works well.

**Environment**
- OS: ubuntu 18.04
- Compiler: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

**To Reproduce**
I made test.proto file like below 

```
syntax = "proto3";

message AfTestSmall
{
    int32 v0 = 1;
    int32 v1 = 2;
    int32 v2 = 3;
    int32 v3 = 4;
}
```

and install package and run protoc
```
./vcpkg install protobuf-c[tools]
./installed/x64-linux/tools/protobuf/protoc --c_out=. --plugin=/home/dev/vcpkg/installed/x64-linux/tools/protoc-gen-c test.proto
```

then fail with message below
```
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
--c_out: protoc-gen-c: Plugin killed by signal 6.
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix crash issue when using tool `protoc-gen-c`：
```
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
--c_out: protoc-gen-c: Plugin killed by signal 6.
```

Fixes: #10745.

Thanks @lipnitsk !

### CODE REVIEW

**Comment by JackBoosY** [comment]
@victech-yeoinbo Could you test this PR?

Thanks.

**Comment by victech-yeoinbo** [comment]
Yeah. I tested this PR and it worked well.
Thanks!

**Review by PhoebeHui — APPROVED** [review]
LGTM

File: ports/protobuf-c/fix-usage-issue.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
++
+ SET(PACKAGE_STRING "${PACKAGE_NAME} ${PACKAGE_VERSION}")
+ ADD_DEFINITIONS(-DPACKAGE_VERSION="${PACKAGE_VERSION}")
+ ADD_DEFINITIONS(-DPACKAGE_STRING="${PACKAGE_STRING}")
+@@ -67,6 +72,9 @@ FILE(GLOB PROTOC_GEN_C_SRC ${MAIN_DIR}/protoc-c/*.h ${MAIN_DIR}/protoc-c/*.cc )
+ ADD_EXECUTABLE(protoc-gen-c ${PROTOC_GEN_C_SRC})
+ 
+ TARGET_LINK_LIBRARIES(protoc-gen-c ${PROTOBUF_PROTOC_LIBRARY} ${PROTOBUF_LIBRARY})
++IF(CMAKE_HOST_UNIX)
++    TARGET_LINK_LIBRARIES(protoc-gen-c ${CMAKE_THREAD_LIBS_INIT})
```
Comment (Neumann-A) [inline ports/protobuf-c/fix-usage-issue.patch:22]: Why not `Threads::Threads`?

Comment (JackBoosY) [reply ports/protobuf-c/fix-usage-issue.patch:22]: @Neumann-A That use the upstream PR [#425](https://github.com/protobuf-c/protobuf-c/pull/425) changes.

**Pull request merged**